internal/types/scope/scope.go

```diff
@@ -38,3 +41,19 @@ var Map = map[string]Type{
 	Org.String():     Org,
 	Project.String(): Project,
 }
+
+// AllowedIn returns the set of Scopes a Resource is allowed in.
+func AllowedIn(r resource.Type) []Type {
+	switch r {
+	case resource.Alias, resource.Billing, resource.Worker:
+		return []Type{Global}
+	case resource.Account, resource.AuthMethod, resource.AuthToken, resource.ManagedGroup, resource.Policy, resource.Scope, resource.SessionRecording, resource.StorageBucket, resource.User:
+		return []Type{Global, Org}
+	case resource.All, resource.Group, resource.Role:
+		return []Type{Global, Org, Project}
+	case resource.CredentialLibrary, resource.Credential, resource.CredentialStore, resource.HostCatalog, resource.HostSet, resource.Host, resource.Session, resource.Target:
+		return []Type{Project}
+	default:
+		return []Type{Unknown}
```

Comment:
What do you think about having this return an error instead? This should only happen when adding a new resource and forgetting to update this function, right? Maybe even a panic 🤔. Returning this might make it harder to track down where the problem is, but let me know if there's something I'm missing!

Reply:
I like the idea of returning an `InvalidParameter` error! Updated this function to return an error: 98fe2786a7cd34ad5426194a4c8424e38ec26851